src/pages/Loans/components/LoanCards/CreateLoanCard/CreateLoanCard.js

```diff
@@ -0,0 +1,208 @@
+
+import { ButtonPrimary } from '../../../../../components/Button';
+import Card from '../../../../../components/Card';
+import { TradeInput } from '../../../../../components/Input';
+import { HeadingSmall } from '../../../../../components/Typography';
+import { getGasInfo, getEthRate, getWalletInfo } from '../../../../../ducks';
+import { createLoan } from '../../../../../ducks/loans';
+import { toggleGweiPopup } from '../../../../../ducks/ui';
+
+import {
+	FormInputRow,
+	FormInputLabel,
+	FormInputLabelSmall,
+	CurrencyKey,
+} from '../../../../../shared/commonStyles';
+
+import { COLLATERAL_PAIR } from '../../../constants';
+
+import NetworkInfo from '../NetworkInfo';
+
+const {
+	ISSUANCE_RATIO,
+	COLLATERAL_CURRENCY_KEY,
+	BORROWED_CURRENCY_KEY,
+	MIN_LOAN_SIZE,
+} = COLLATERAL_PAIR;
+
+import { TxErrorMessage } from '../commonStyles';
+
+export const CreateLoanCard = ({
+	toggleGweiPopup,
+	gasInfo,
+	ethRate,
+	walletInfo: { balances, currentWallet },
+	createLoan,
+}) => {
+	const { t } = useTranslation();
+
+	const [collateralAmount, setCollateralAmount] = useState('');
+	const [borrowedAmount, setBorrowedAmount] = useState('');
+
+	const [gasLimit, setLocalGasLimit] = useState(gasInfo.gasLimit);
+	const [collateralAmountErrorMessage, setCollateralAmountErrorMessage] = useState(null);
+	const [txErrorMessage, setTxErrorMessage] = useState(null);
+
+	const handleSubmit = async () => {
+		const {
+			snxJS: { EtherCollateral },
+			utils,
+		} = snxJSConnector;
+
+		setTxErrorMessage(null);
+
+		try {
+			const openLoanArgs = {
+				value: utils.parseEther(collateralAmount),
+				gasPrice: gasInfo.gasPrice * GWEI_UNIT,
+				gasLimit,
+			};
+
+			const gasEstimate = await EtherCollateral.contract.estimate.openLoan(openLoanArgs);
+			const updatedGasEstimate = Number(gasEstimate);
+			setLocalGasLimit(updatedGasEstimate);
+
+			const tx = await EtherCollateral.openLoan({ ...openLoanArgs, gasLimit: updatedGasEstimate });
+
+			console.log(tx);
+
+			createLoan({
+				collateralAmount,
+			});
+		} catch (e) {
+			setTxErrorMessage(t('common.errors.unknown-error-try-again'));
+		}
+	};
+
+	const showGweiPopup = () => toggleGweiPopup(true);
+
+	const collateralCurrencyBalance = getCurrencyKeyBalance(balances, COLLATERAL_CURRENCY_KEY);
+	const borrowedCurrencyBalance = getCurrencyKeyBalance(balances, BORROWED_CURRENCY_KEY);
+
+	useEffect(() => {
+		setCollateralAmountErrorMessage(null);
+		if (collateralAmount != '') {
+			if (currentWallet && collateralAmount > collateralCurrencyBalance) {
+				setCollateralAmountErrorMessage(t('common.errors.amount-exceeds-balance'));
+			} else if (collateralAmount < MIN_LOAN_SIZE) {
+				setCollateralAmountErrorMessage(
+					t('loans.loan-card.errors.min-loan-size', {
+						currencyKey: COLLATERAL_CURRENCY_KEY,
+						minLoanSize: MIN_LOAN_SIZE,
+					})
+				);
+			}
+		}
+	}, [collateralAmount, collateralCurrencyBalance, t, currentWallet]);
```

Comment:
do you need all of these? Might trigger useEffect for no reasons.

If you wanted to silent the eslint error maybe just use `// eslint-disable-next-line react-hooks/exhaustive-deps`

Reply:
Yeah all of these are used